### PR TITLE
Add leopard react

### PR DESCRIPTION
**Pull request by albho**
will need to change after npm publish & before merging
- `@picovoice/leopard-react` in `/demo/react/package.json`
- uncomment last line in `/binding/react/README.md`

### CODE REVIEW

**Comment by albho** [comment]
will need to change after npm publish & before merging
- `@picovoice/leopard-react` in `/demo/react/package.json`
- uncomment last line in `/binding/react/README.md`

File: binding/react/README.md
Comment (ksyeo1010) [inline binding/react/README.md:27]: This is `LeopardWeb` description. You should take a look at Porcupine/Rhino's react readme

File: binding/react/README.md
Comment (ksyeo1010) [inline binding/react/README.md:134]: you should put options here. I think other animals don't do anything with the options so we mention anything.

File: binding/react/README.md
Comment (ksyeo1010) [inline binding/react/README.md:173]: mention that states goes to false.

File: binding/react/README.md
Comment (ksyeo1010) [inline binding/react/README.md:174]: should merge the two model sections into 1.

File: binding/react/package.json
Comment (ksyeo1010) [inline binding/react/package.json:3]: 1.2.0

File: binding/react/src/use_leopard.ts
Comment (ksyeo1010) [inline binding/react/src/use_leopard.ts:2]: 2023 only

File: demo/react/src/VoiceWidget.tsx
Comment (ksyeo1010) [inline demo/react/src/VoiceWidget.tsx:5]: has to be imported like this?

File: demo/react/README.md
Comment (ksyeo1010) [inline demo/react/README.md:9]: should mention about file demo as well.

File: demo/react/src/VoiceWidget.tsx
Comment (ksyeo1010) [inline demo/react/src/VoiceWidget.tsx:185]: add seconds so it is a bit more clear

File: demo/react/src/VoiceWidget.tsx
Comment (albho) [reply demo/react/src/VoiceWidget.tsx:5]: yes, web-voice-processor doesn't export `PvEngine`. Could just remove this and let react do type inference

**Review by ksyeo1010 — APPROVED** [review]

File: binding/react/package.json
Comment (ksyeo1010) [inline binding/react/package.json:45]:
```suggestion
    "@picovoice/web-voice-processor": "~4.0.8",
```

File: binding/react/src/use_leopard.ts
Comment (ksyeo1010) [inline binding/react/src/use_leopard.ts:96]: set error if no leopardRef.

File: binding/react/package.json
Comment (ksyeo1010) [inline binding/react/package.json:45]: wanan do a yarn so yarn lock is updated

**Review by ksyeo1010 — APPROVED** [review]

File: binding/react/src/use_leopard.ts
Comment (laves) [inline binding/react/src/use_leopard.ts:192]: Unless I'm missing something, it looks as though we aren't checking if we're currently recording or not before we try to start recording again. Generally, you'd want to check if we're recording and then either return right away (because there's nothing to do) or throw an error if we want to be more harsh.

File: binding/react/src/use_leopard.ts
Comment (laves) [inline binding/react/src/use_leopard.ts:165]: was this meant to just wrap the `clearInterval` or was it intended to be the mark of whether we are recording or not? You do want to check the recording state, and if we're not recording, return without doing anything.

File: binding/react/README.md
Comment (laves) [inline binding/react/README.md:180]:
```suggestion
**Note**: By default, Leopard will only record for 2 minutes before stopping and processing the buffered audio. This is to prevent unbounded memory usage. To increase this limit, call `startRecording` with the optional `maxRecordingSec` parameter:
```

File: demo/react/src/VoiceWidget.tsx
Comment (laves) [inline demo/react/src/VoiceWidget.tsx:74]: can you add some margins between the buttons? they're a bit mushed together

File: demo/react/src/VoiceWidget.tsx
Comment (laves) [inline demo/react/src/VoiceWidget.tsx:63]: since initialization and transcription for Leopard take a significant amount of time, you are going to want to disable buttons that you don't want the user to press while it's busy. For instance, I can click initialize and start recording one after another and cause an error.

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/react/README.md
Comment (laves) [reply binding/react/README.md:221]: This is still and issue

**Review by laves — APPROVED** [review]

**Review by laves — APPROVED** [review]

**Pull request merged**
